src/service/RaceGameService.js

```diff
@@ -0,0 +1,74 @@
+import Car from '../model/Car.js';
+import GAME_STATUS from '../constants/gameStatus.js';
+import { MissionUtils } from '@woowacourse/mission-utils';
+import { MOVE_DETERMINING_NUMBER, RANDOM_NUMBER } from '../constants/Config.js';
+
+class RaceGameService {
+  #status;
+  #cars;
+  #maxRound;
+  #currentRound;
+  #onEndRound;
+  #onBeforeStart;
+
+  constructor () {
+    this.#status = GAME_STATUS.CREATED;
+    this.#currentRound = 0;
+  }
+
+  start () {
+    this.#onBeforeStart();
+    while (this.#currentRound < this.#maxRound) {
+      this.#currentRound += 1;
+      this.playRound();
+    }
+    this.#status = GAME_STATUS.END;
+  }
+
+  ready (carNames, maxRound, onEndRound, onBeforeStart) {
+    this.#status = GAME_STATUS.READY;
+    const cars = carNames.map((name) => new Car(name));
+    this.#cars = cars;
+    this.#maxRound = maxRound;
+    this.#onBeforeStart = onBeforeStart;
+    this.#onEndRound = onEndRound;
+  }
+
+  playRound () {
+    this.#status = GAME_STATUS.PLAYING;
+    for (const car of this.#cars) {
+      if (this.#canMove()) {
+        car.move();
+      }
+    }
```

Comment:
forEach를 사용하면 라인수는 그대로지만
반복문의 시작과 끝을 인지한 상태에서 반복문 내용을 읽는다 => 한 줄로된 함수를 읽는다
로 가독성이 좋아지지 않을까요? 😊

```suggestion
    this.#cars.forEach((car) => {
      if (this.#canMove()) {
        car.move();
      }
    });
```

Reply:
오우. 좋네요. 👍🏻👍🏻👍🏻습관적으로 이터레이터를 돌려버렸네요..!